### PR TITLE
x11: use swapchain and expose age

### PR DESCRIPTION
**Pull request by Drakulix**
Builds upon #419 and #420 to bring buffer age support to the x11 backend as well.

### CODE REVIEW

**Comment by i509VCB** [comment]
This is great, but a couple of questions:

1. Given the swapchain change, we may want to consider just flattening `Present` away into x11 surface.
2. Maybe mark the function to acquire the buffer as `#[must_use] `?

**Comment by i509VCB** [comment]
I have tested this, everything appears to work as intended.

**Comment by Drakulix** [comment]
> This is great, but a couple of questions:
> 
>     1. Given the swapchain change, we may want to consider just flattening `Present` away into x11 surface.
> 
>     2. Maybe mark the function to acquire the buffer as `#[must_use]`?

Yes, those are probably some good suggestions.

File: anvil/src/x11.rs [hunk 1]
```diff
@@ -328,10 +322,9 @@ pub fn run_x11(log: Logger) {
             window.set_cursor_visible(cursor_visible);
 
             // Send frame events so that client start drawing their next frame
-            state
-                .window_map
-                .borrow()
-                .send_frames(start_time.elapsed().as_millis() as u32);
+            window_map.send_frames(start_time.elapsed().as_millis() as u32);
+            std::mem::drop(window_map);
```
Comment (i509VCB) [inline anvil/src/x11.rs:326]: This seems a bit odd to put in the busy loop?

Comment (Drakulix) [reply anvil/src/x11.rs:326]: It is necessary for the next line, because otherwise state is borrowed.

Comment (i509VCB) [reply anvil/src/x11.rs:326]: Oh it's probably because window map is in a ref cell

Comment (i509VCB) [reply anvil/src/x11.rs:326]: Oh it's probably because window map is in a ref cell

File: anvil/src/x11.rs [hunk 2]
```diff
@@ -204,122 +204,116 @@ pub fn run_x11(log: Logger) {
...
+
+                        Ok(())
+                    },
+                )
+                .map_err(Into::<SwapBuffersError>::into)
+                .and_then(|x| x)
+                .map_err(Into::<SwapBuffersError>::into)
+            {
+                Ok(()) => {
+                    // Unbind the buffer and now let the scope end to present.
```
Comment (i509VCB) [inline anvil/src/x11.rs:303]: There is no scope to drop anymore as we submit on the next line.

**Pull request merged**